### ISSUE TITLE
Can't choose other db schema than public in supabaseDataProvider

### ISSUE THREAD

**Issue by spyriaev** [marmelab/ra-supabase#31]
I can't change a schema different from the `public` one passing a parameterized supbaseClient to supabaseDataProvider.

```typescript
const options: SupabaseClientOptions<"stats"> = {
    db: { schema: 'stats' }
}
export const supbaseClient =
    createClient<StatsDatabase>(process.env.NEXT_PUBLIC_SUPABASE_URL!,
        process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!,
        options);

```

Would you mind considering to use generic parameters in supabaseDataProvider?

**Comment by djhi** [comment]
Sure, can you open a PR?

**Comment by slax57** [comment]
Fixed via #47
